CONTRIBUTING.md

```diff
@@ -61,17 +61,21 @@ You can also use the regular `dune build @check` etc... targets.
 
 #### Nix
 
-We have a Nix flake that you can use. For development, simply run `nix develop`.
+We have a Nix flake that you can use. For development, simply run `nix develop` for server development.
+otherwise for client development we expose separate shells, e.g client-vscode so `nix develop .#client-vscode`.
 
-If you wish to do `nix build` or compose this flake from another project, you
+you can view the list of packages and devShells exposed by `nix flake show`
+
+If you wish to do `nix build`, you
 will need to use the .?submodules=1` trick, since we use submodules here for
 vendoring. For example, building requires:
 
 ```
 nix build .?submodules=1
 ```
 
-You can use this flake in other flakes or Nix derivations.
+This currently only applies to building the default package (coq-lsp), which is the server. 
+Clients don't have specific submodules as of yet.
```

Comment:
Rather than submodules, I would say shells. I mainly wanted the flake for development however if we wish to distribute OCaml pacakges relying on Coq libraries the situation is a bit complicated. We also want people to be able to override the serapi and coq dependencies and avoid using submodules. This is not easy to do from what I have tried.

Reply:
I don't believe submodules and shells are referring to the same thing here?


> We also want people to be able to override the serapi and coq dependencies and avoid using submodules. This is not easy to do from what I have tried.

 I couldn't even get it to build without coq vendored in with submodules, overriding serapi worked fine, but coq wasn't even building.